### PR TITLE
Add more flexibility to headers

### PR DESCRIPTION
**Pull request by geggleto**
Would like to see an optional separator as sometimes I don't want to use commas. Kind of nit-picking... but would be nice.

### CODE REVIEW

File: proposed/http-message.md [hunk 1]
```diff
@@ -399,7 +399,7 @@ interface MessageInterface
      * @param string $name Case-insensitive header field name.
      * @return string|null
      */
-    public function getHeader($name);
+    public function getHeader($name, $separator = ',');
```
Comment (GrahamCampbell) [inline proposed/http-message.md]: this needs to be added to the phpdoc too

**Comment by GrahamCampbell** [comment]
:-1: on the idea

**Comment by weierophinney** [comment]
:-1:

We specifically chose to only allow `,` to reduce the number of possible implementation paths. If you want to implement an alternate separator, you can retrieve all values for a header and pass the value to `implode()`. Once #493/#494 are merged, this will look like:

```php
$concatenated = implode(';', $message->getHeader($name));
```

**Comment by mtdowling** [comment]
:-1:

**Comment by weierophinney** [comment]
@pmjones and/or @simensen — please close.

**Pull request closed without merging**